src/ripple/app/paths/PathRequest.cpp

```diff
@@ -585,10 +604,20 @@ PathRequest::findPaths (std::shared_ptr<RippleLineCache> const& cache, int const
         }
     }
 
+    // Charge cost is based on source currencies used.
+    double const min_cost = 50;
+    double const max_cost = 400;
+    Resource::Charge c {static_cast<int>(
+        std::min(max_cost, std::max(
+            min_cost, std::pow(sourceCurrencies.size(), 2)))),
+                "path update"};
+    consumer_.charge(c);
+
```

Comment:
@seelabs Just fyi, your `static_assert(maxCost * maxCost < ...` is also ok.  The compiler does the overflow checking for you, and if it overflows, the compiler whines about the expression not being an integral constant expression (hard compile-time error).